pandas/tests/series/test_ufunc.py

```diff
@@ -255,8 +255,8 @@ def __add__(self, other):
         pd.array([1, 3, 2], dtype="int64"),
         pd.array([1, 10, 0], dtype="Sparse[int]"),
         pd.to_datetime(["2000", "2010", "2001"]),
-        pd.to_datetime(["2000", "2010", "2001"]).tz_localize("CET"),
-        pd.to_datetime(["2000", "2010", "2001"]).to_period(freq="D"),
+        pd.to_datetime(["2000", "2010", "2001"]).tz_localize("CET"),  # type: ignore
+        pd.to_datetime(["2000", "2010", "2001"]).to_period(freq="D"),  # type: ignore
```

Comment:
```
$ mypy pandas
pandas\tests\series\test_ufunc.py:258: error: "DatetimeIndex" has no attribute "tz_localize"
pandas\tests\series\test_ufunc.py:259: error: "DatetimeIndex" has no attribute "to_period"
Found 2 errors in 1 file (checked 1018 source files)
```

Reply:
IIRC this happens bc we use inherit_names on DatetimeIndex.  I'd be OK with pinning the inherited methods in a way mypy gets even if its a little more verbose if you think its worthwhile